packages/ember-metal/lib/binding.js

```diff
@@ -169,19 +162,45 @@ Binding.prototype = {
   connect(obj) {
     assert('Must pass a valid object to Ember.Binding.connect()', !!obj);
 
-    var fromPath = this._from;
-    var toPath = this._to;
-    trySet(obj, toPath, getWithGlobals(obj, fromPath));
+    let fromObj, fromPath;
+
+    // If the binding's "from" path could be interpreted as a global, verify
+    // whether the path refers to a global or not by consulting `Ember.lookup`.
+    if (isGlobalPath(this._from)) {
+      let name = getFirstKey(this._from);
+      let possibleGlobal = Ember.lookup[name];
+
+      if (possibleGlobal) {
+        deprecate(
+          `Connecting an Ember.Binding to a global path such as '${this._from}' is deprecated. Please use a service instead.`,
+          false,
+          { id: 'ember-metal.global-bindings', until: '3.0.0' }
```

Comment:
Can you please make an entry in the [deprecation guide](https://github.com/emberjs/website/blob/master/source/deprecations/v2.x.html.md) and add a `url` property here.